custom_components/tuya_local/goldair_portable_airconditioner.yaml

```diff
@@ -0,0 +1,126 @@
+name: 4-in-1 Portable Air Conditioner
+products:
+  - id: gylvqvz3p94i3zdn
+    name: Goldair Portable Air Conditioner
+primary_entity:
+  entity: climate
+  dps:
+    - id: 1
+      type: boolean
+      name: hvac_mode
+      mapping:
+        - dps_val: false
+          value: "off"
+        - dps_val: true
+          constraint: mode
+          conditions:
+            - dps_val: cold
+              value: cool
+            - dps_val: hot
+              value: heat
+            - dps_val: wet
+              value: dry
+            - dps_val: wind
+              value: fan_only
+            - dps_val: heat
+              value: heat
+
+    - id: 2
+      type: integer
+      name: temperature
+      range:
+        min: 16
+        max: 31
+      unit: C
+    - id: 3
+      type: integer
+      name: current_temperature
+      # 107, 108 listed as F temperatures in report, but not existing
+      # in log, and there is no temperature_unit dp to work with
+    - id: 4
+      type: string
+      name: mode
+      hidden: true
+    - id: 5
+      type: string
+      name: fan_mode
+      mapping:
+        - dps_val: low
+          value: low
+        - dps_val: med
+          value: medium
+        - dps_val: high
+          value: high
+    - id: 15
+      type: string
+      name: swing_1
```

Comment:
Should this be combined with swing-mode for horizontal and both? It seems to be labelled swingmodeH in the info you put in the description.

Reply:
From the "Query Things Data Model" results I can see 15 is called Vertical swing wind.  I tested 15 dps just now and it goes beep but does nothing. This model only does one swing mode. Tuya calls 110 Horizontal swing but in real life it's actually a vertical swing as it goes up and down, not left to right. So basically Tuya has incorrectly labeled horizontal and vertical and they need to be switched and dps 15 ignored. Shall I delete 15? There is one portable ac model above this one that perhaps would support both swing modes and ionize.

Reply:
If I remove the non relevant dps from the yaml then when I add it to Tuya Local it shows the Nedis AC first as it matches more dps. Is there a preferred way to reference them without them actually showing up.
I tried 
```
secondary_entities:
  - entity: switch
    translation_key: ionizer
    # this model doesn't support ion
    dps:
      - id: 11
        type: boolean
        name: switch
        hidden: true
        optional: true
```
        
but it still showed up.

Reply:
Hey thanks for your help getting this improved. I've just tested the most up to date version that you've done and it's working really nicely.
![image](https://github.com/user-attachments/assets/a4978a3d-a839-49a9-b0fe-d2e190180b02)

However I'm not quite sure if the Swing mode is conditionally displaying correctly. 
When I set Swing mode to On it then immediately changes to Horizontal, as does Both mode. I suppose because On and Horizontal are the same thing and Both is also triggering Horizontal. All three modes do enable swinging on the unit.

I have some other AC's that aren't Tuya based and they do support both swing modes and this is how they show up
![image](https://github.com/user-attachments/assets/2af01c51-3505-49f8-ab23-4235858bda5b)

From what I can gather for this particular Goldair Portable AC it should only show Off and Horizontal (as strictly by Tuya dps values). On & both should be hidden? Vertical is correctly hidden. I would think Both should only show if Vertical is also supported. Reading through the YAML I can't yet identify where the issue is. I can take a much closer look on Monday

Reply:
I was trying to make it show "off" and "on" when only the horizontal setting was available. But I think I got tied in knots trying to cover all the possibilities with the available logic.  It seems that masking two bits at once makes them an "or" operation, rather than an "and", so I can't use that.  Maybe I will just give up trying to cover the cases of the "horizontal" swing not being available to simplify the logic.

Reply:
I've tested the newest code, and it's almost perfect. Now it only shows swing mode as Off or On. When I select Off or On the unit responds correctly. However in HA almost straight after I click On it updates to display as 'swing mode' and has in logs
```
2024-12-02 21:34:48.082 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [140136051543776] Swing mode  is not valid. Valid swing modes are: on, off
2024-12-02 21:35:03.760 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [140136051543776] Swing mode  is not valid. Valid swing modes are: on, off
```
![image](https://github.com/user-attachments/assets/4ba04a85-03e0-40a3-a8a5-025f151efc23)